### ISSUE TITLE
Lint control attributes (allow/deny/etc) have no effect on generic parameters

### ISSUE THREAD

**Issue by petrochenkov** [rust-lang/rust#61552]
```rust
fn f<#[allow(warnings)] foo>() {} // warning: type parameter `foo` should have an upper camel case name
```
Expected behavior - no warning, actual behavior - warning.

cc https://github.com/rust-lang/rust/issues/61238, which is a similar issue

**Comment by istankovic** [comment]
Seems to be fixed. Tested with 1.73.

**Comment by fmease** [comment]
> Seems to be fixed. Tested with 1.73.

Only for type parameters. The issue is still present for lifetime and const parameters, it seems.

Type parameters (works):

```rs
//@ check-pass
#![deny(non_camel_case_types)]
struct S<#[allow(non_camel_case_types)] t>(t);
```

Lifetime parameters (doesn't work):

```rs
#![deny(non_snake_case)]
struct Z<#[allow(non_snake_case)] 'A>(&'A ()); //~ ERROR
```

Const parameters (doesn't work):

```rs
#![deny(non_upper_case_globals)]
struct S<#[allow(non_upper_case_globals)] const n: usize>; //~ ERROR
```

**Comment by fmease** [comment]
I did a cursory reading through various parts like `<LateContext<'_> as LintContext>::opt_span_lint` (...), `LateContextAndPass`, various visitors, etc. but nothing stood out. It's especially weird that it now works for type params but not for the other two kinds of generic params given the fact that they are unified under `{ast,hir}::GenericParam`.
